app/components/goodcity/add-item-overlay.js

```diff
@@ -80,15 +82,17 @@ export default Ember.Component.extend({
   },
 
   actions: {
-    moveItemToBox() {
-      let pkg = this.get("pkg");
+    async moveItemToBox(pkg, cb) {
       if (pkg) {
         if (this.hasInvalidAddedQuantity()) {
           this.get("messageBox").alert(
             this.get("i18n").t("box_pallet.invalid_quantity")
           );
         } else {
-          this.resolveAddItemPromises();
+          await this.resolveAddItemPromises();
+          if (cb) {
+            cb(pkg);
+          }
         }
```

Comment:
```suggestion
    async moveItemToBox(pkg, cb = _.noop) {
      if (!pkg) return;
      if (this.hasInvalidAddedQuantity()) {
        return this.get("messageBox").alert(
          this.get("i18n").t("box_pallet.invalid_quantity")
        );
      }
     
     await this.resolveAddItemPromises();
     cb(pkg);
     return pkg; // return it anyways, that way you support both callbacks and promises
  }
```

Untested, but something like that would be an improvement. You can support both callbacks and promises. 
Tip: using `_.noop` as a default value removes the need for an `if {}` statement

Also `resolveAddItemPromises` is a bit confusing...not sure what an "addItemPromise" is 🤔

Reply:
Yeah its a bit difficult to understand it. Would love to rewrite this file, but it just increases the scope. So I kept it as it is for now